### PR TITLE
Add decorator classes

### PR DESCRIPTION
**Pull request by BranBayou**
### In this project I did the following
- created a class Nameable and implement a method called correct_name that will raise a NotImplementedError.
- I made sure Person class inherits from Nameable.
- created base decorator that inherits from Nameable.
- in the constructor I assigned a nameable object from params to an instance variable.
- implemented the correct_name method that returns the result of the correct_name method of the @nameable.
- For the CapitalizeDecorator:
   created a class that inherits from the base Decorator class.
   implemented a method correct_name that capitalizes the output of @nameable.correct_name.
- For the TrimmerDecorator:
   created a class that inherits from the base Decorator class.
   implemented a method correct_name that makes sure that the output of @nameable.correct_name has a maximum of 10 characters. If it's longer it should trim the word.
- saw my decorators in action
- I checked for linter errors.

### CODE REVIEW

File: base_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+require './nameable'
+
+class Decorator
```
Comment (muneebulrehman) [inline base_decorator.rb]: You have required `./nameable` but you have not inherited from it. `Decorator` class should be the subclass of `Nameable`. Kindly do so.

**Review by muneebulrehman — CHANGES_REQUESTED** [review]
Hi @BranBayou , 

Good job so far! I see you put a lot of effort into this project. However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- No linter errors
- Descriptive PR
- Great job adding all the decorator classes.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: base_decorator.rb (hunk 1 above)
Comment (BranBayou) [reply base_decorator.rb]: Hi @muneebulrehman,
I fixed the required changes.

**Review by VICTOR-CHUKWU — APPROVED** [review]
Hi @BranBayou , 
# STATUS: APPROVED! 🥇 💯 
done

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- [ ] All linters passed
- [ ] classes well decorated

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**